trip-schema/src/main/java/global/maplink/trip/schema/v1/exception/TripErrorType.java

```diff
@@ -8,14 +8,27 @@
 @AllArgsConstructor
 public enum TripErrorType implements ValidationViolation {
     ROUTE_POINTS_LESS_THAN_TWO("The route must contain at least two site points"),
+    VARIABLE_AXLES_FROMSITEID_EMPTY("fromSiteId should not be empty"),
+    VARIABLE_AXLES_TOSITEID_EMPTY("toSiteId should not be empty"),
+    VARIABLE_AXLES_FROMSITEID_POINTING_TO_LAST_SITE("fromSiteId cannot be equal to the last site in trip problem"),
+    VARIABLE_AXLES_FROMSITEID_SAME_AS_TOSITEID("fromSiteId and toSiteId cannot be the same: "),
+    VARIABLE_AXLES_OVERLAP_FOUND("VariableAxles leg overlap found on "),
+    MSG_SITEID_NOT_FOUND_IN_PROBLEM("siteId not found in trip problem sites: "),
+    MSG_CONTAINED_IN("does not have a valid value. Allowed values: "),
     PROFILE_NAME_EMPTY("Field profileName should not be empty"),
     CALCULATION_MODE_FIELD_EMPTY("Field calculationMode should not be empty"),
     CALLBACK_DOES_NOT_HAVE_URL("Callback parameters must have an url"),
     TOLL_PARAMETERS_DOES_NOT_HAVE_VEHICLE_TYPE("Toll parameters must have a vehicleType"),
     CROSSED_BORDERS_DOES_NOT_HAVE_LEVEL("CrossedBorders parameters must have a level"),
     ROAD_TYPE_ELEMENTS_EMPTY("SpeedPreferences should contain one element for each roadType"),
     VEHICLE_SPECIFICATION_CONTAINS_NEGATIVE_VALUE("VehicleSpecification should not contain any measure with a negative value"),
+    TEMPORARY(""),
     TRIP_PROFILE_NOT_FOUND("Trip profile does not exist");
 
-    private final String message;
+    private String message;
+
+    public TripErrorType addToMessage(String textToadd){
+        TEMPORARY.message = this.message + textToadd;
```

Comment:
isso é muito perigoso, valores de enum devem ser imutáveis, se não condições concorrentes podem quebrar completamente, já que um enum é singleton a nivel de aplicação.

Reply:
Eu também achava que "de preferência" valores de enum devem ser final, mas, segundo o Bealdung pode ter alguns casos ..."Second, our label field is final. While fields of an enum do not have to be final, in most cases we don't want our labels to change...."  eheheh. Mas concordo, só que não sei como resolver esse problema descrito no comentário acima para poder concatenar textos variáveis na mensagem do enum e continuar devolvendo um enum de TripErrorType.

Reply:
Se um campo não é final, ele pode ser editado a nível de aplicação, tem ideia de como isso é perigoso?

Uma ação de um usuario impactar outros usuários, já que a constante é shared.

Ainda por cima, estariamos introduzindo esse tipo de bug dentro da aplicação dos clientes.

Enums pode ter campos não final assim como campos static pode ser não finals, em casos muito raros, em aplicações muito específicas, e esse não é um desses casos.